public/agg_types/comparing_controller.js

```diff
@@ -1,11 +1,37 @@
 import _ from 'lodash'; // TODO: refactor lodash dependencies
+import { handleCustomDate } from './lib/custom_date_handler';
 
 export function comparingAggController($scope) {
 
   $scope.isCustomComparing = () => {
     return $scope.agg.params.range.comparing.display === 'Custom';
   };
 
+  $scope.hasDifferentRanges = () => {
+    const customComparingTexts = $scope.agg.params.range.custom;
+    if (!$scope.isCustomComparing() || !customComparingTexts) return false;
+
+    // Checks if input texts are not empty
+    const isNullOrEmpty = text => text == null || text === '' || text && text.trim() === '';
```

Comment:
`text === ''` is useless. It's already covered by the last clause

Reply:
In this case, this validation is needed since last clause will return false for empty strings:
`!!'' && ''.trim() === ''` -> `false`

But I think it could be rewritten like this (without `text &&` in last clause, since we're already checking `null` or `empty` in the first one):
`const isNullOrEmpty = text => text == null || text.trim() === '';`

Reply:
You're right! Great!